### DISCUSSION TITLE
Querying version from a plain website which redirects to S3 - Databricks

### DISCUSSION THREAD

**Discussion by pat-s** [renovatebot/renovate#24687] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.108.0

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I am trying to implement a `customDatasource` following https://docs.renovatebot.com/modules/datasource/custom/#plain via the `plain` type.

I am trying to support versioning for the Databricks JDBC driver. It does not have a repo or similar available, only the following download site: https://www.databricks.com/spark/jdbc-drivers-archive

This site again has links to S3 buckets which result in a final version scheme of something like `2.6.26.1045`.

### What you tried so far.

I've played around for some time but can't wrap my head around it. How can I transform the plain response into JSON which is then properly read by `renovate`? Is this possible at all? I saw the `customDatasource` feature being relatively new.

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "customManagers": [
    {
      "customType": "regex",
      "fileMatch": ["^Dockerfile$"],
      "depNameTemplate": "DatabricksOdbcDriver",
      "datasourceTemplate": "custom.DatabricksOdbcDriver",
      "matchStrings": [
        "datasource=(?<datasource>.*?) registryUrl=(?<registryUrl>.*?)( format=(?<format>.*?))?\\sARG .*?_VERSION=(?<currentValue>.*)\\s"
      ],
      "versioningTemplate": "regex:^(?<major>\\d+\\.)(?<minor>\\d+\\.)(?<patch>.*)$"
    }
  "customDatasources": {
    "DatabricksOdbcDriver": {
      "defaultRegistryUrlTemplate": "https://www.databricks.com/spark/jdbc-drivers-archive",
      "format": "plain",
      "transformTemplates": [FIXME]
    }
  }
```

Dockerfile

```
# renovate: datasource=custom registryUrl=https://www.databricks.com/spark/jdbc-drivers-archive format=plain
ARG DATABRICKS_ODBC_DRIVER=2.6.26.1045
```

### Relevant debug logs

-

**Comment by secustor** [comment, marked as answer]
The page you are using is not type plain, but rather a HTML page. 
https://docs.renovatebot.com/modules/datasource/custom/#plain

You will have to find a JSON or PLAIN endpoint for now to solve this, tough there is an open PR which would allow you to extract the available versions from download links
https://github.com/renovatebot/renovate/pull/24403

**Reply by pat-s** [reply]
Ahh thanks. I just thought that plain is the generic fallback if json is not available. Will look into the PR!
